media/reference/main.js

```diff
@@ -0,0 +1,401 @@
+        switch (message.type) {
+            case 'setChapters':
+                {
+                    setChapters(message.data);
+
+                    break;
+                }
+            case 'setChapterImage':
+                {
+                    setChapterImage(message.data.chapter, message.data.image);
+
+                    break;
+                }
+            case 'setFunctions':
+                {
+                    setFunctions(
+                        message.data.chapter, 
+                        message.data.chapterTitle, 
+                        message.data.chapterImage,
+                        message.data.groups,
+                        message.data.localGroups,
+                        message.data.functions,
+                        message.data.categoryImages);
+
+                    break;
+                }
+            case 'setCategoryImage':
+                {
+                    setCategoryImage(message.data.category, message.data.image);
+
+                    break;
+                }
+            case 'setFunctionDetails':
+                {
+                    setFunctionDetails(message.data.chapter, message.data.details, message.data.categoryImages);
+
+                    break;
+                }
+        }
+    });
+
+    setInterval(keepAlive, 1000);
+
+    vscode.postMessage({ type: 'webViewReady' });
+
+    function keepAlive() {
+        vscode.postMessage({ type: 'webViewKeepAlive' });
+    }
+
+    function setChapters(chapters) {
+        const content = document.querySelector('.reference-content');
+
+        if (content) {
+            content.textContent = '';
+
+            for (const chapter of chapters) {
+                const chapterNameValue = chapter['name'];
+                const chapterTitleValue = chapter['title'];
+                const chapterSubtitleValue = chapter['subtitle'];
+                const chapterImageValue = chapter['image'];
+                
+                const chapterDiv = document.createElement('div');
+                chapterDiv.className = 'chapter-button';
+                chapterDiv.onclick = function() {
+                    vscode.postMessage({ type: 'getFunctions', chapter: chapterNameValue });
+                }
+                content.appendChild(chapterDiv)
+
+                const chapterImageDiv = document.createElement('div');
+                chapterImageDiv.className = 'chapter-image-container';
+                chapterDiv.appendChild(chapterImageDiv);
+
+                const chapterImage = document.createElement('img');
+                chapterImage.id = `${chapterNameValue}-image`;
+                if (chapterImageValue) {
+                    chapterImage.src = `data:image/png;base64,${chapterImageValue}`;
+                }
+                else {
+                    chapterImage.src = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=";
```

Comment:
Placeholder image?

Reply:
Yeah, it's a single transparent pixel so we don't see the "broken image" icon.